--- interceptor.c.orig	2008-01-10 22:19:29.000000000 +0100
+++ interceptor.c	2008-01-10 22:19:52.000000000 +0100
@@ -711,8 +711,18 @@
 
         break;
     case CNI_DISCARD:
+    	/* patch found on ubuntuforums.org, written by aranoyas */
+    	/* override local LAN access */
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(2,6,14)
+	rc2 = original_ip_handler.orig_handler_func(skb, dev, type, dev);
+#else
+	rc2 = original_ip_handler.orig_handler_func(skb, dev, type);
+#endif
+
+	/*
         dev_kfree_skb(skb);
         rx_dropped++;
+	*/
         break;
     default:
         printk(KERN_DEBUG "RECV: Unhandled case in %s rc was %x\n",
@@ -833,8 +843,13 @@
         /* packet dropped */
         else
         {
+	    /* patch found on ubuntuforums.org, originally written by aranoyas */
+	    /* override local LAN access */
+	    rc2 = pBinding->InjectSend(skb, dev);
+	    /*
             dev_kfree_skb(skb);
             tx_dropped++;
+	    */
         }
         break;
     case CNI_CHAIN:
