--- a/dist/broker.js	2019-08-09 11:49:55.288015620 +0200
+++ b/dist/broker.js	2019-08-09 14:24:46.813741507 +0200
@@ -698,7 +698,7 @@
             if (!this.wsToUUIDs.get(ws)) {
                 this.wsToUUIDs.set(ws, {});
             }
-            this.wsToUUIDs.get(ws)[uuid] = true;
+            // this.wsToUUIDs.get(ws)[uuid] = true;
         }
         if (ws && !this.wsToKeys.get(ws)) {
             this.wsToKeys.set(ws, {});

