diff --git a/src/main/radiusd.c b/src/main/radiusd.c
index 0b35d76ef5..e72cc257f9 100644
--- a/src/main/radiusd.c
+++ b/src/main/radiusd.c
@@ -419,6 +419,13 @@ int main(int argc, char *argv[])
 	/*
 	 *  Disconnect from session
 	 */
+
+#  ifdef HAVE_SYSTEMD
+	if (sd_notify(0, "READY=1") <=0) {
+		WARN("Something goes wrong on send READY event to systemd");
+	}
+#  endif
+
 	if (main_config.daemonize) {
 		pid_t pid;
 		int devnull;
@@ -480,9 +487,6 @@ int main(int argc, char *argv[])
 				exit(EXIT_FAILURE);
 			}
 
-#  ifdef HAVE_SYSTEMD
-			sd_notify(0, "READY=1");
-#  endif
 
 			exit(EXIT_SUCCESS);
 		}
