diff --git a/src/modules/rlm_python3/rlm_python3.c b/src/modules/rlm_python3/rlm_python3.c
index 9f78011e35..12b5fc447d 100644
--- a/src/modules/rlm_python3/rlm_python3.c
+++ b/src/modules/rlm_python3/rlm_python3.c
@@ -485,6 +485,8 @@ static rlm_rcode_t do_python_single(REQUEST *request, PyObject *pFunc, char cons
 	int		ret;
 	int 		i;
 
+	if (!pFunc) return RLM_MODULE_NOOP;
+
 	/* Default return value is "OK, continue" */
 	ret = RLM_MODULE_OK;
 
