--- empathy-3.5.90/configure.ac	2012-09-01 16:24:05.684454939 +0200
+++ empathy-3.5.90b/configure.ac	2012-09-01 16:23:54.137787349 +0200
@@ -226,7 +226,7 @@
    telepathy-farstream >= $TELEPATHY_FARSTREAM_REQUIRED
    clutter-1.0 >= $CLUTTER_REQUIRED
    clutter-gtk-1.0 >= $CLUTTER_GTK_REQUIRED
-   clutter-gst-1.0 >= $CLUTTER_GST_REQUIRED
+   clutter-gst-2.0 >= $CLUTTER_GST_REQUIRED
 ])
 
 AC_DEFINE(EMPATHY_GOA_PROVIDER, "org.gnome.OnlineAccounts",
