diff --git a/pyanaconda/ui/gui/hubs/progress.glade b/pyanaconda/ui/gui/hubs/progress.glade
index cabd07f74..39d743166 100644
--- a/pyanaconda/ui/gui/hubs/progress.glade
+++ b/pyanaconda/ui/gui/hubs/progress.glade
@@ -98,6 +98,7 @@ Finish it and then click the Finish configuration button please.</property>
                                 <property name="label" translatable="yes" context="GUI|Progress">_Finish configuration</property>
                                 <property name="use_action_appearance">False</property>
                                 <property name="visible">True</property>
+                                <property name="sensitive">False</property>
                                 <property name="can_focus">True</property>
                                 <property name="receives_default">True</property>
                                 <property name="halign">end</property>
@@ -151,6 +152,7 @@ Go ahead and reboot to start using it!</property>
                                 <property name="label" translatable="yes" context="GUI|Progress">_Reboot</property>
                                 <property name="use_action_appearance">False</property>
                                 <property name="visible">True</property>
+                                <property name="sensitive">False</property>
                                 <property name="can_focus">True</property>
                                 <property name="receives_default">True</property>
                                 <property name="halign">end</property>
diff --git a/pyanaconda/ui/gui/hubs/progress.py b/pyanaconda/ui/gui/hubs/progress.py
index 3f915137c..e21c0e827 100644
--- a/pyanaconda/ui/gui/hubs/progress.py
+++ b/pyanaconda/ui/gui/hubs/progress.py
@@ -132,6 +132,7 @@ class ProgressHub(Hub):
 
         GLib.source_remove(self._rnotes_id)
         self._progressNotebook.set_current_page(1)
+        self.window.set_may_continue(True)
 
         iutil.ipmi_report(IPMI_FINISHED)
 
diff --git a/pyanaconda/ui/gui/hubs/summary.glade b/pyanaconda/ui/gui/hubs/summary.glade
index 88313c248..447d4eec5 100644
--- a/pyanaconda/ui/gui/hubs/summary.glade
+++ b/pyanaconda/ui/gui/hubs/summary.glade
@@ -24,9 +24,6 @@
                 <child>
                   <placeholder/>
                 </child>
-                <child>
-                  <placeholder/>
-                </child>
               </object>
             </child>
           </object>
@@ -100,6 +97,7 @@
                 <property name="label" translatable="yes" context="GUI|Summary">_Begin Installation</property>
                 <property name="use_action_appearance">False</property>
                 <property name="visible">True</property>
+                <property name="sensitive">False</property>
                 <property name="can_focus">True</property>
                 <property name="receives_default">True</property>
                 <property name="use_underline">True</property>
@@ -127,8 +125,8 @@
             <property name="can_focus">False</property>
             <property name="margin_right">6</property>
             <property name="margin_bottom">6</property>
-            <property name="xalign">1</property>
             <property name="label" translatable="yes">We won't touch your disks until you click 'Begin Installation'.</property>
+            <property name="xalign">1</property>
             <attributes>
               <attribute name="font-desc" value="Cantarell Italic"/>
               <attribute name="foreground" value="#808080808080"/>
