diff --git a/package.json b/package.json
index 2d38432..8d16508 100644
--- a/package.json
+++ b/package.json
@@ -17,7 +17,7 @@
     "@types/node": "^9.4.4",
     "mocha": "*",
     "prettier": "^1.18.2",
-    "typescript": "^2.7.1"
+    "typescript": "^5.5.4"
   },
   "scripts": {
     "prepublish": "npm run build",
@@ -33,4 +33,4 @@
   },
   "author": "ForbesLindesay",
   "license": "MIT"
-}
\ No newline at end of file
+}
diff --git a/src/binaryOperation.ts b/src/binaryOperation.ts
index 99244d5..be0c891 100644
--- a/src/binaryOperation.ts
+++ b/src/binaryOperation.ts
@@ -23,7 +23,7 @@ export type Operator =
   | '<=';
 
 export default function binaryOperation(
-  operator: Operator,
+  operator: any,
   left: any,
   right: any,
 ): any {
diff --git a/src/index.ts b/src/index.ts
index bdc51be..a68cd7a 100644
--- a/src/index.ts
+++ b/src/index.ts
@@ -16,7 +16,7 @@ export function expressionToConstant(
   options: ExpressionToConstantOptions = {},
 ): {constant: true; result: any} | {constant: false; result?: void} {
   let constant = true;
-  function toConstant(expression: b.Expression): any {
+  function toConstant(expression: any): any {
     if (!constant) return;
     if (b.isArrayExpression(expression)) {
       const result = [];
