--- zabbix-2.4.5/configure.ac	2015-04-22 09:56:19.000000000 +0200
+++ zabbix-2.4.5.b/configure.ac	2015-06-11 11:07:25.460706132 +0200
@@ -1072,9 +1072,6 @@
 	dnl Checking for IBM DB2 support
 	AX_LIB_IBM_DB2()
 	if test "x$want_ibm_db2" = "xyes"; then
-		if test "x$have_db" != "xno"; then
-			AC_MSG_ERROR([You can configure for only one database.])
-		fi
 
 		if test "x$found_ibm_db2" = "xyes"; then
 			have_db="IBM DB2"
@@ -1087,9 +1084,6 @@
 	dnl Checking for MySQL support
 	AX_LIB_MYSQL()
 	if test "x$want_mysql" = "xyes"; then
-		if test "x$have_db" != "xno"; then
-			AC_MSG_ERROR([You can configure for only one database.])
-		fi
 
 		if test "x$found_mysql" = "xyes"; then
 			have_db="MySQL"
@@ -1102,9 +1096,6 @@
 	dnl Checking for Oracle support
 	AX_LIB_ORACLE_OCI([10.0])
 	if test "x$WANT_ORACLE_OCI" = "xyes"; then
-		if test "x$have_db" != "xno"; then
-			AC_MSG_ERROR([You can configure for only one database.])
-		fi
 
 		if test "x$HAVE_ORACLE_OCI" = "xyes"; then
 			have_db="Oracle"
@@ -1126,9 +1117,6 @@
 	dnl Checking for PostgreSQL support
 	AX_LIB_POSTGRESQL("8.1")
 	if test "x$want_postgresql" = "xyes"; then
-		if test "x$have_db" != "xno"; then
-			AC_MSG_ERROR([You can configure for only one database.])
-		fi
 
 		if test "x$found_postgresql" = "xyes"; then
 			if test "$postgresql_version_check" != "1"; then
@@ -1149,9 +1137,6 @@
 	dnl Checking for SQLite3 support
 	AX_LIB_SQLITE3()
 	if test "x$WANT_SQLITE3" = "xyes"; then
-		if test "x$have_db" != "xno"; then
-			AC_MSG_ERROR([You can configure for only one database.])
-		fi
 
 		if test "x$found_sqlite3" = "xyes"; then
 			have_db="SQLite v3.x"
