diff -BNaur xorg-server-1.13.0-orig/xfixes/xfixes.c xorg-server-1.13.0/xfixes/xfixes.c
--- xorg-server-1.13.0-orig/xfixes/xfixes.c	2012-11-19 00:56:35.200344612 +0100
+++ xorg-server-1.13.0/xfixes/xfixes.c	2012-11-19 01:18:07.123738714 +0100
@@ -151,8 +151,6 @@
 
     if (pXFixesClient->major_version >= NUM_VERSION_REQUESTS)
         return BadRequest;
-    if (stuff->xfixesReqType > version_requests[pXFixesClient->major_version])
-        return BadRequest;
     return (*ProcXFixesVector[stuff->xfixesReqType]) (client);
 }
 
