commit 05d6caacd3aa53a3125675511dca3ea9269357fa
Author: Daniele Rondina <geaaru@sabayonlinux.org>
Date:   Tue Jun 25 23:34:05 2019 +0200

    Add update of expire_date on image

diff --git a/lxd/images.go b/lxd/images.go
index 8342a9e4..3a83ea80 100644
--- a/lxd/images.go
+++ b/lxd/images.go
@@ -1578,6 +1578,12 @@ func imagePatch(d *Daemon, r *http.Request) Response {
 		info.Public = public
 	}
 
+	// Get Expiry
+	expiry, err := reqRaw.GetInt64("expiry_date")
+	if err == nil {
+		info.ExpiresAt = time.Unix(expiry, 0)
+	}
+
 	// Get Properties
 	_, ok := reqRaw["properties"]
 	if ok {
diff --git a/shared/json.go b/shared/json.go
index 09f10666..51268950 100644
--- a/shared/json.go
+++ b/shared/json.go
@@ -40,6 +40,16 @@ func (m Jmap) GetInt(key string) (int, error) {
 	}
 }
 
+func (m Jmap) GetInt64(key string) (int64, error) {
+    if val, ok := m[key]; !ok {
+		return -1, fmt.Errorf("Response was missing `%s`", key)
+	} else if val, ok := val.(int64); !ok {
+		return -1, fmt.Errorf("`%s` was not an int64", key)
+	} else {
+		return int64(val), nil
+	}
+}
+
 func (m Jmap) GetBool(key string) (bool, error) {
 	if val, ok := m[key]; !ok {
 		return false, fmt.Errorf("Response was missing `%s`", key)
