Index: exmap-0.10/kernel/Makefile
===================================================================
--- exmap-0.10.orig/kernel/Makefile
+++ exmap-0.10/kernel/Makefile
@@ -7,7 +7,7 @@ obj-m += exmap.o
 build: kernel_modules
 
 kernel_modules:
-	make -C /lib/modules/$(shell uname -r)/build M=$(CURDIR) modules
+	$(MAKE) -C /lib/modules/$(shell uname -r)/build M=$(CURDIR) modules
 
 clean:
-	make -C /lib/modules/$(shell uname -r)/build M=$(CURDIR) clean
+	$(MAKE) -C /lib/modules/$(shell uname -r)/build M=$(CURDIR) clean
Index: exmap-0.10/Makefile
===================================================================
--- exmap-0.10.orig/Makefile
+++ exmap-0.10/Makefile
@@ -19,4 +19,4 @@ clean: $(SUBDIRS)
 	$(DOSUBDIRS)
 
 test: build
-	make -C src test
+	$(MAKE) -C src test
