Index: exmap-0.10/jutil/Makefile
===================================================================
--- exmap-0.10.orig/jutil/Makefile
+++ exmap-0.10/jutil/Makefile
@@ -13,7 +13,7 @@ OBJ += $(ETOBJ)
 EXES += egtest
 
 CXX=g++
-CXXFLAGS=-Wall -Werror -g -I.
+CXXFLAGS += -Wall -g -I.
 LD=g++
 LDFLAGS=-lpcre
 
Index: exmap-0.10/src/Makefile
===================================================================
--- exmap-0.10.orig/src/Makefile
+++ exmap-0.10/src/Makefile
@@ -11,7 +11,7 @@ JUTILDIR=../jutil
 
 EXMAP_OBJ=Exmap.o Range.o Elf.o
 
-CXXFLAGS += -g -Wall -Werror -I$(JUTILDIR)
+CXXFLAGS += -g -Wall -I$(JUTILDIR)
 LDFLAGS += -lpcre -ljutil -L$(JUTILDIR)
 
 GTKCXXFLAGS = `pkg-config --cflags gtkmm-2.4`
