--- a/gio/gdbus-2.0/codegen/gdbus-codegen.in	2013-01-19 00:17:12.000000000 +0100
+++ b/gio/gdbus-2.0/codegen/gdbus-codegen.in	2013-02-24 10:30:27.541809387 +0100
@@ -29,13 +29,12 @@
 
 if srcdir is not None:
     path = os.path.join(srcdir, 'gio', 'gdbus-2.0')
-elif os.name == 'nt':
-    # Makes gdbus-codegen 'relocatable' at runtime on Windows.
-    path = os.path.join(os.path.dirname(__file__), '..', 'lib', 'gdbus-2.0')
+    sys.path.insert(0, os.path.abspath(path))
+    from codegen import codegen_main
 else:
     path = os.path.join('@datadir@', 'glib-2.0')
 
 sys.path.insert(0, os.path.abspath(path))
-from codegen import codegen_main
+from gdbus_codegen import codegen_main
 
 sys.exit(codegen_main.codegen_main())
