--- ZoneMinder-1.24.2.orig/src/zm_thread.cpp	2009-06-07 13:55:44.000000000 +0200
+++ ZoneMinder-1.24.2/src/zm_thread.cpp	2010-03-28 18:34:52.623391708 +0200
@@ -26,6 +26,7 @@
 #include <signal.h>
 #include <errno.h>
 #include <sys/time.h>
+#include <stdio.h>
 
 struct timespec getTimeout( int secs )
 {
--- ZoneMinder-1.24.2.orig/src/zm_utils.cpp	2009-05-28 10:47:59.000000000 +0200
+++ ZoneMinder-1.24.2/src/zm_utils.cpp	2010-03-28 18:34:27.874329163 +0200
@@ -21,6 +21,7 @@
 #include "zm_utils.h"
 
 #include <stdarg.h>
+#include <stdio.h>
 
 const std::string stringtf( const char *format, ... )
 {
