diff --git a/dvbplayer.c b/dvbplayer.c
index b0adac4..a3b1d1d 100644
--- a/dvbplayer.c
+++ b/dvbplayer.c
@@ -291,6 +291,8 @@ cDvbPlayer::cDvbPlayer(const char *FileName, bool PauseLive)
      delete index;
      index = NULL;
      }
+  else if (PauseLive)
+     framesPerSecond = cRecording(FileName).FramesPerSecond(); // the fps rate might have changed from the default
 }
 
 cDvbPlayer::~cDvbPlayer()
