http://article.gmane.org/gmane.linux.vdr/45847

diff -pruN vdr-1.7.27-orig/channels.c vdr-1.7.27/channels.c
--- vdr-1.7.27-orig/channels.c  2012-03-25 22:46:14.294920001 -0700
+++ vdr-1.7.27/channels.c       2012-03-28 09:29:23.734920001 -0700
@@ -955,7 +955,7 @@ int cChannels::MaxShortChannelNameLength
   if (!maxShortChannelNameLength) {
      for (cChannel *channel = First(); channel; channel = Next(channel)) {
          if (!channel->GroupSep())
-            maxShortChannelNameLength = max(Utf8StrLen(channel->ShortName()), maxShortChannelNameLength);
+            maxShortChannelNameLength = max(Utf8StrLen(channel->ShortName(true)), maxShortChannelNameLength);
          }
      }
   return maxShortChannelNameLength;
