diff -Naur vdr-1.7.32.orig/Makefile vdr-1.7.32/Makefile
--- vdr-1.7.32.orig/Makefile	2012-11-20 19:29:00.000000000 +0100
+++ vdr-1.7.32/Makefile	2012-11-20 19:30:26.000000000 +0100
@@ -206,11 +206,11 @@
 
 install-dirs:
 	@mkdir -p $(DESTDIR)$(VIDEODIR)
-	@mkdir -p $(DESTDIR)$(CONFDIRDEF)
 	@mkdir -p $(DESTDIR)$(CACHEDIRDEF)
 	@mkdir -p $(DESTDIR)$(RESDIRDEF)
 
 install-conf:
+	@mkdir -p $(DESTDIR)$(CONFDIRDEF)
 	@cp -n *.conf $(DESTDIR)$(CONFDIRDEF)
 
 
