commit 560dd544010bebef5d03434771b5f69c3083f92b
Author: Vincent Rogier <vince.rogier@gmail.com>
Date:   Fri Dec 14 12:27:49 2018 +0100

    fix #188

diff --git a/src/column.c b/src/column.c
index 6b36c2f..9f452b9 100644
--- a/src/column.c
+++ b/src/column.c
@@ -261,6 +261,8 @@ boolean OCI_ColumnDescribe
                 col->props |=  OCI_CPF_IS_GEN_BY_DEFAULT_ON_NULL;
             }
 
+        #if OCI_VERSION_COMPILE >= OCI_18_1
+
             if (value & OCI_ATTR_COL_PROPERTY_IS_LPART)
             {
                 col->props |= OCI_CPF_IS_LPART;
@@ -270,6 +272,8 @@ boolean OCI_ColumnDescribe
             {
                 col->props |= OCI_CPF_IS_CONID;
             }
+
+        #endif 
         }
     }
 
