diff -urN nspluginwrapper-0.9.90.orig/src/npw-config.c nspluginwrapper-0.9.90/src/npw-config.c
--- nspluginwrapper-0.9.90.orig/src/npw-config.c	2005-10-26 21:50:05.000000000 +0100
+++ nspluginwrapper-0.9.90/src/npw-config.c	2005-12-01 14:30:15.000000000 +0000
@@ -80,7 +80,7 @@
 
 static const char *get_system_mozilla_plugin_dir(void)
 {
-  return LIBDIR "/mozilla/plugins";
+  return LIBDIR "/nsbrowser/plugins";
 }
 
 static const char *get_user_mozilla_plugin_dir(void)
@@ -98,8 +98,9 @@
 static const char **get_mozilla_plugin_dirs(void)
 {
   static const char *default_dirs[] = {
-	"/usr/lib/mozilla/plugins",
-	"/usr/lib64/mozilla/plugins"
+	"/usr/lib/nsbrowser/plugins",
+	"/usr/lib32/nsbrowser/plugins",
+	"/usr/lib64/nsbrowser/plugins"
   };
 
   const int n_default_dirs = sizeof(default_dirs) / sizeof(default_dirs[0]);
